### PR TITLE
Add systemd@v2 gatherer documentation

### PR DESCRIPTION
**Pull request by arbulu89**
`systemd@v2` gatherer documentation.
Based on: https://github.com/trento-project/agent/pull/278

### CODE REVIEW

File: guides/gatherers.md [hunk 1]
```diff
@@ -1094,6 +1095,62 @@ Example output (in Rhai):
 "inactive";
 ```
 
+### systemd@v2
+
+**Argument required**: yes.
+
+Gather systemd units information. It returns an object with multiple fields about the systemd unit.
+
+The provided service name must include the extension, such as `.service` or `.mount`.
```
Comment (rtorrero) [inline guides/gatherers.md:1104]: "The provided service name" -> "The provided unit name"

File: guides/gatherers.md [hunk 2]
```diff
@@ -1094,6 +1095,62 @@ Example output (in Rhai):
...
+
+Only a subset of properties are returned. The information about them is available in
```
Comment (rtorrero) [inline guides/gatherers.md:1106]: "The information about them" -> "Additional information about these"

**Review by rtorrero — APPROVED** [review]
LGTM, two minor improvements proposed

**Comment by rtorrero** [comment]
@arbulu89 after a bit of thought, might it make sense to maybe point out the differences between the v1 and v2? so someone can easily spot what changed?

**Comment by arbulu89** [comment]
> I just noticed something wrong, the index points to `#systemdv2` but the section is called `systemd@v2` which prevents the linking from working AFAICS

Did you try @rtorrero ?
I did, and this is the unique way that the link works. It was not a typo at least, I wrote that deliberately

**Comment by rtorrero** [comment]
> > I just noticed something wrong, the index points to `#systemdv2` but the section is called `systemd@v2` which prevents the linking from working AFAICS
> 
> Did you try @rtorrero ? I did, and this is the unique way that the link works. It was not a typo at least, I wrote that deliberately

I've tested with a few different editors and you are actually right, in GH flavored MD (most common I guess), the symbols are simply removed from the links

**Review by rtorrero — APPROVED** [review]

**Comment by arbulu89** [comment]
@rtorrero (and others). Based on the markdown chapters linking conversation, I have tried with the `ex_doc` docs, and unfortunately the linking fails there...
How github and hex create the links for the chapters is different.
I didn't find any universal solution for all of them.
One think we could do, is ti name the chapter as `systemd - v2` instead of using the `@` which is the troublemaker

**Comment by arbulu89** [comment]
Following up the last comment, the last change I did adding the `<span id="systemd-v2"></span>` makes the job.
This tag tricks the markdown page to move to the correct position in both github and hex docs.
Not a elegant solution, but we have both documentations at least working, and it is not really invasive.
Opinions?

**Comment by CDimonaco** [comment]
For me the important thing is the documentation tbh, markdown allows html tags exactly for this edge cases, for me it's the most correct way @arbulu89

**Review by EMaksy — APPROVED** [review]
Hey the docs are looking good :D

**Review by CDimonaco — APPROVED** [review]

**Pull request merged**